utils/find_corpus.py

```diff
@@ -128,6 +128,186 @@ def get_sacrebleu(source: str, target: str):
+    """
+    Returns monolingual datasets ordered by size. Datasets with few downloads are ignored
+    as they are probably low quality and not trustworthy.
+    """
+    from huggingface_hub import DatasetFilter, HfApi
+
+    api = HfApi()
+
+    datasets = list(
+        api.list_datasets(
+            filter=DatasetFilter(
+                #
+                language=language,
+                multilinguality="monolingual",
+            )
+        )
+    )
+    datasets.sort(key=lambda dataset: -dataset.downloads)
+    datasets.sort(key=lambda dataset: -HF_DATASET_SIZES.get(get_size(dataset.tags), 0))
+
+    print("")
+    print("┌─────────────────────────────────────────────────┐")
+    print("│ huggingface monolingual data                    │")
+    print("└─────────────────────────────────────────────────┘")
+    print_table(
+        [
+            ["ID", "Size", "Downloads"],
+            *[
+                [
+                    #
+                    f"https://huggingface.co/datasets/{dataset.id}",
+                    get_size(dataset.tags),
+                    dataset.downloads,
+                ]
+                for dataset in datasets
+                if is_useful_dataset(dataset)
+            ],
+        ]
+    )
+
+
+def get_huggingface_parallel(source: str, target: str):
+    """
+    Returns parallel datasets ordered by size. Datasets with few downloads are ignored
+    as they are probably low quality and not trustworthy.
+    """
+    from huggingface_hub import DatasetFilter, HfApi
+
+    api = HfApi()
+
+    datasets = list(
+        api.list_datasets(
+            filter=DatasetFilter(
+                #
+                language=[source, target],
+            )
+        )
+    )
+    datasets.sort(key=lambda dataset: -dataset.downloads)
+    datasets.sort(key=lambda dataset: -HF_DATASET_SIZES.get(get_size(dataset.tags), 0))
+
+    print("")
+    print(
+        "┌────────────────────────────────────────────────────────────────────────────────────────────────────┐"
+    )
+    print(
+        f"│ huggingface parallel data https://huggingface.co/datasets?language=language:{source},language:{target}"
+    )
+    print(
+        "└────────────────────────────────────────────────────────────────────────────────────────────────────┘"
+    )
+    print_table(
+        [
+            ["ID", "Size", "Downloads"],
+            *[
+                [
+                    #
+                    f"https://huggingface.co/datasets/{dataset.id}",
+                    get_size(dataset.tags),
+                    dataset.downloads,
+                ]
+                for dataset in datasets
+                if is_useful_dataset(dataset)
+            ],
+        ]
+    )
+
+
+def is_useful_dataset(dataset: any) -> bool:
+    """Determines if a dataset is useful or not."""
+    return (
+        "task_categories:automatic-speech-recognition" not in dataset.tags
+        and dataset.downloads > 5
```

Comment:
this one is a bit subjective. Can we just add the number of downloads to the table and let the user decide on this? I can imagine someone uploaded a fresh dataset for low resource language that has < 5 downloads

Reply:
Sure, most of them were junk, but I can leave them unfiltered.